include/spdlog/sinks/basic_file_sink-inl.h

```diff
@@ -26,6 +26,11 @@ SPDLOG_INLINE const filename_t &basic_file_sink<Mutex>::filename() const {
     return file_helper_.filename();
 }
 
+template <typename Mutex>
+SPDLOG_INLINE void basic_file_sink<Mutex>::truncate() {
+    file_helper_.reopen(true);
```

Comment:
This should be locked using the mutex in the base class. See other sinks for example code.

Reply:
Gosh, sorry for that. Updated.